src/commands/serve.rs

```diff
@@ -199,10 +201,74 @@ async fn download_init_package(url: Option<&str>) -> std::io::Result<()> {
     Ok(())
 }
 
+pub async fn clear_cache(req: actix_web::HttpRequest) -> actix_web::HttpResponse {
+    LOCK.write().await;
+    fpm::apis::cache::clear(&req).await
+}
+
+// TODO: Move them to routes folder
+async fn sync(
+    req: actix_web::web::Json<fpm::apis::sync::SyncRequest>,
+) -> actix_web::Result<actix_web::HttpResponse> {
+    LOCK.write().await;
+    fpm::apis::sync(req).await
+}
+
+async fn sync2(
+    req: actix_web::web::Json<fpm::apis::sync2::SyncRequest>,
+) -> actix_web::Result<actix_web::HttpResponse> {
+    LOCK.write().await;
+    fpm::apis::sync2(req).await
+}
+
+pub async fn clone() -> actix_web::Result<actix_web::HttpResponse> {
+    LOCK.read().await;
```

Comment:
All these statements are wrong. The lock is released when the https://docs.rs/async-lock/latest/async_lock/struct.RwLockReadGuard.html is dropped. So write:

```rust
let _l = LOCK.read().await;
```

So lock is kept till end of block.